src/test/java/com/emeq/counter/api/FoodsApiTest.java

```diff
@@ -0,0 +1,178 @@
+        JsonNode json = mapper.createObjectNode();
+        ((ObjectNode) json).put("id", 1);
+        ((ObjectNode) json).put("name", fruits[0]);
+
+        final ResponseEntity<JsonNode> response = this.restTemplate.getForEntity(endpoint("/foods/1"), JsonNode.class);
+
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+        assertThat(response.getBody().toString()).isEqualTo(json.toString());
+    }
+
+    @Test
+    public void createFood() {
+        JsonNode request = mapper.createObjectNode();
+        ((ObjectNode) request).put("name", "Banana");
+
+        JsonNode json = mapper.createObjectNode();
+        ((ObjectNode) json).put("id", 7);
+        ((ObjectNode) json).put("name", "Banana");
+
+        ResponseEntity<JsonNode> response = this.restTemplate.postForEntity(endpoint("/foods"), request, JsonNode.class);
+
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.CREATED);
+        assertThat(response.getHeaders().getLocation().getPath()).isEqualTo("/foods/7");
+        assertThat(response.getBody().toString()).isEqualTo(json.toString());
+    }
+
+    @Test
+    public void updateSuccess() throws URISyntaxException {
+        JsonNode requestJson = mapper.createObjectNode();
+        ((ObjectNode) requestJson).put("name", "Watrmelon");
+
+        JsonNode json = mapper.createObjectNode();
+        ((ObjectNode) json).put("id", 6);
+        ((ObjectNode) json).put("name", "Watrmelon");
+
+        RequestEntity<JsonNode> requestSuccess = RequestEntity
+                .put(new URI(endpoint("/foods/6")))
+                .headers(valid_content_type())
+                .body(requestJson);
+
+        ResponseEntity<JsonNode> responseSuccess = this.restTemplate.exchange(requestSuccess, JsonNode.class);
+
+        assertThat(responseSuccess.getStatusCode()).isEqualTo(HttpStatus.OK);
+        assertThat(responseSuccess.getBody().toString()).isEqualTo(json.toString());
+    }
+
+    @Test
+    public void updateNotFound() throws URISyntaxException {
+        JsonNode requestJson = mapper.createObjectNode();
+        ((ObjectNode) requestJson).put("name", "Watrmelon");
+
+        RequestEntity<JsonNode> requestNotFound = RequestEntity.put(new URI(endpoint("/foods/10")))
+                .headers(valid_content_type()).body(requestJson);
+
+        ResponseEntity<JsonNode> responseNotFound = this.restTemplate.exchange(requestNotFound, JsonNode.class);
+
+        assertThat(responseNotFound.getStatusCode()).isEqualTo(HttpStatus.NOT_FOUND);
+    }
+
+    @Test
+    public void deleteSuccessFood() throws URISyntaxException {
+        RequestEntity<Void> request = RequestEntity
+                .delete(new URI(endpoint("/foods/6")))
+                .headers(valid_content_type())
+                .build();
+        ResponseEntity<Object> responseSuccess = this.restTemplate.exchange(request, Object.class);
+
+        assertThat(responseSuccess.getStatusCode()).isEqualTo(HttpStatus.NO_CONTENT);
```

Comment:
I think you should also assert that food is gone. Either getting it from show action or directly accessing repository by `findById` etc. should suffice.